pkg/ingester/ingester.go

```diff
@@ -717,8 +727,25 @@ func (i *Ingester) Label(ctx context.Context, req *logproto.LabelRequest) (*logp
 		}
 	}
 
+	allValues := listutil.MergeStringLists(resp.Values, storeValues)
+
+	if req.Values {
```

Comment:
Can you write a test for this, I think there's a bug when there's `Values` and no filter.

Reply:
Indeed there was a bug there. Tests were added.